docs/RelayServer.md

```diff
@@ -0,0 +1,240 @@
+contains a description of the behaviour that is problematic:
+
+> Note, at present, snapshotting does not place all of the block or receipt data
+> into the database. This means you will not get information relating to
+> transactions more than a few days old.
+
+So, please double check that warp synchronization is turned off when using
+parity. The whole trustlines system may appear to work, but will give wrong
+results if you run with warp mode enabled.
+
+### contracts
+The [trustlines-contracts
+repository](https://github.com/trustlines-network/contracts) contains the
+solidity contracts to be deployed on the blockchain. The ('how to deploy the
+contracts guide')[] <!-- XXX --> contains more information on how to deploy the
+contracts. We assume from now on that the contracts have already been deployed
+and that the `addresses.json` file has been copied to the users home directory.
+
+<!-- XXX Does that even get created while deploying the contracts? -->
+
+### PostgreSQL
+The trustlines system uses a PostgreSQL database to store some user data and to
+keep a synchronized view on the relevant state from the blockchain.
+
+On an ubuntu based system this PostgreSQL can be installed with
+```
+sudo apt install postgresql
+```
+
+After that a database user must be created. Please use a real password:
+```
+sudo -u postgres psql <<EOF
+CREATE USER trustlines WITH PASSWORD 'choose-a-password-here';
+ALTER USER trustlines CREATEDB;
+EOF
+```
+
+After that the postgres environment should be configured for the account running
+the relay server:
+```
+echo >>~/.pgpass 'localhost:5432:*:trustlines:choose-a-password-here'; chmod 0600 ~/.pgpass
+cat >>~/.bashrc <<EOF
+export PGUSER=trustlines
+export PGHOST=localhost
+export PGDATABASE=trustlinesdb
+EOF
+```
+
+After logging in again in order to have the environment variables set, the user
+should be able to create the database:
+
+```
+createdb trustlinesdb
+```
+
+
+The rest of this tutorial assumes that the user running the relay and
+py-eth-index related commands has a working postgresql environment configured.
+All programs consider the PG* environment variables and will read ~/.pgpass for
+information about passwords. The [py-eth-index section](#py-eth-index) describes
+how to create the trustlines specific tables.
+
+### py-eth-index
+The [py-eth-index repository](https://github.com/trustlines-network/py-eth-index)
+contains a helper program that synchronizes the relevant information from the
+blockchain into a postgresql database.
+
+#### Installation of py-eth-index
+
+Clone the git repository:
+```
+cd ~
+git clone https://github.com/trustlines-network/py-eth-index
+```
+We need a virtualenv for this repository, so let's create it:
+```
+python3 -m venv ~/opt/py-eth-index; ~/opt/py-eth-index/bin/pip install -U pip
+```
+and install py-eth-index
+```
+cd ~/py-eth-index
+~/opt/py-eth-index/bin/pip install -c constraints.txt -r requirements.txt
```

Comment:
Why do you always specify the full path and not use the venv by activating it? I have never seen that. I find it confusing, also because the location is confusing to me. I have never seen someone putting the venv not in the home folder.

Reply:
Well, I use ~/opt as some kind of user-writable /opt filesystem here (see https://www.tldp.org/LDP/Linux-Filesystem-Hierarchy/html/opt.html for more information on /opt). 
~/opt lives in the home folder. I am not aware of any location where users should put their virtualenvs. I guess there's no standard location to put them. spotify's dh-virtualenv puts them into /opt.

Activating the virtualenv gives more room to make errors, e.g. you could run one of the pip install commands in the wrong virtualenv (essentially destroying it, since we do have conflicting requirements). 
Though we could create symlinks in ~/bin/ to the ethindex and tl-relay programs, that won't hurt.

Reply:
ah sorry, I actually did not see that you use ~/opt, I always read /opt.
In the home folder is fine. 
I just think it is distracting to always write ~/opt/env/bin/ethindex and then hard to see what is actually happening.
Just writting ethindex seems to me more clear that we now call the program.